tests/testthat/test-derive_params_growth_age.R

```diff
@@ -1,49 +1,478 @@
+
+  vs_sds <- vs_data_meta %>% mutate(AVAL = (((VSSTRESN / M)^L) - 1) / (L * S))
+  vs_pctl <- vs_sds %>% mutate(AVAL = pnorm(AVAL) * 100)
+  expected <- bind_rows(vs_sds, vs_pctl) %>% pull(AVAL)
+
+
+  expect_equal(
+    filter(actual, PARAMCD %in% c("BMIASDS", "BMIAPCTL")) %>% pull(AVAL),
+    expected
+  )
+})
+
 
-## Test 1: derive_params_growth_age works ----
-test_that("derive_params_growth_age Test 1: derive_params_growth_age works", {
+
+## Test 4: Head circumference derivation works ----
+test_that("derive_params_growth_age Test 4: Head circumference SDS and percentile works", {
   vs_data <- tibble::tribble(
     ~STUDYID, ~USUBJID, ~VISIT, ~SEX, ~AGECUR, ~AGEU, ~VSTESTCD, ~VSSTRESN,
-    "STUDY", "1001", "1", "F", 24.5, "months", "WEIGHT", 10,
-    "STUDY", "1002", "1", "F", 25.49, "months", "WEIGHT", 11,
-    "STUDY", "1002", "2", "F", 26.7, "months", "WEIGHT", 11.5,
-    "STUDY", "1003", "1", "F", 25.51, "months", "WEIGHT", 12,
-    "STUDY", "1004", "1", "F", 27.5, "months", "WEIGHT", 13
+    "Study", "1001", "Screening", "M", 61, "days", "HEADC", 39,
+    "Study", "1002", "Cycle 1 Day 1", "F", 1157, "days", "HEADC", 50,
   )
 
-  fake_meta <- tibble::tribble(
+  meta <- tibble::tribble(
     ~SEX, ~AGE, ~AGEU, ~L, ~M, ~S,
-    "F", 24.5, "months", 1, 2, 3,
-    "F", 25.5, "months", 4, 5, 6,
-    "F", 27, "months", 7, 8, 9,
-    "F", 28, "months", 10, 11, 12
+    "M", 61, "days", 1, 39.1349, 0.02997,
+    "F", 1157, "days", 1, 48.6732, 0.02906,
+  )
+
+  actual <- derive_params_growth_age(
+    dataset = vs_data,
+    by_vars = exprs(STUDYID, USUBJID, VISIT),
+    sex = SEX,
+    age = AGECUR,
+    age_unit = AGEU,
+    meta_criteria = meta,
+    parameter = VSTESTCD == "HEADC",
+    analysis_var = VSSTRESN,
+    set_values_to_sds = exprs(
+      PARAMCD = "HDCASDS"
+    ),
+    set_values_to_pctl = exprs(
+      PARAMCD = "HDCAPCTL"
+    )
+  )
+
+  vs_data_age <- vs_data %>%
+    mutate(
+      AGECUR_c = case_when(
+        AGEU == "days" ~ AGECUR,
+        AGEU == "months" ~ round(AGECUR * 30.4375)
+      ),
+      AGEU_c = "days"
+    ) %>%
+    mutate(AGE = AGECUR_c, AGEU = AGEU_c) %>%
+    select(-ends_with("_c"), -AGECUR)
+
+  vs_data_meta <- vs_data_age %>% dplyr::inner_join(meta, by = c("SEX", "AGE", "AGEU"))
+
+  vs_sds <- vs_data_meta %>% mutate(AVAL = (((VSSTRESN / M)^L) - 1) / (L * S))
+  vs_pctl <- vs_sds %>% mutate(AVAL = pnorm(AVAL) * 100)
+  expected <- bind_rows(vs_sds, vs_pctl) %>% pull(AVAL)
+
+  expect_equal(
+    filter(actual, PARAMCD %in% c("HDCASDS", "HDCAPCTL")) %>% pull(AVAL),
+    expected
+  )
+})
+
+## Test 5: Extreme BMI value derivation works ----
+test_that("derive_params_growth_age Test 5: Extreme BMI value derivation works", {
+  vs_data <- tibble::tribble(
+    ~STUDYID, ~USUBJID, ~VISIT, ~SEX, ~AGECUR, ~AGEU, ~VSTESTCD, ~VSSTRESN,
+    "Study", "1001", "Screening", "M", 1233, "days", "BMI", 19,
+    "Study", "1002", "Screening", "M", 1233, "days", "BMI", 100,
```

Comment:
@rossfarrugia this is the extended bmi example with the correction in place, including the setting of Z-scores to 8.21 when the percentile is extremely high